code/player/TTTPlayer.Confirmation.cs

```diff
@@ -114,6 +136,21 @@ private void BecomePlayerCorpseOnServer(Vector3 force, int forceBone)
                 Rotation = Rotation
             };
 
+            corpse.KillerWeapon = LastDamageWeapon?.Name;
+            corpse.WasHeadshot = LastDamageWasHeadshot;
+            corpse.Distance = LastDistanceToAttacker;
+            corpse.Suicide = LastAttacker == this;
+
+            List<string> perks = new();
+            PerksInventory perksInventory = (Inventory as Inventory).Perks;
+
+            for (int i = 0; i < perksInventory.Count(); i++)
+            {
+                perks.Add(perksInventory.Get(i).Name);
+            }
+
+            corpse.Perks = perks.ToArray();
```

Comment:
This logic could be cleaned up.

Reply:
Good catch, feel free changing it as I don't get time doing so until tomorrow.